lib/collection/variable-scope.js

```diff
@@ -98,6 +98,25 @@ _.assign(VariableScope.prototype, /** @lends VariableScope.prototype */ {
     },
 
     /**
+     * Using this function, one can sync the values of this variable list from an existing VariableScope JSON.
+     *
+     * @param {Object|VariableScope} obj A VariableScope instance, or it's JSON equivalent.
+     * @param {Boolean=} [track] A flag that indicates whether or not to track synced variables.
+     *
+     * @returns {Object}
+     */
+    syncVariablesFromScope: function (obj, track) {
+        VariableScope.isVariableScope(obj) && (obj = obj.toJSON());
```

Comment:
I would rather love a faster flow. for var scope, I am doing the following in runtime...

- for each variable in source, I am updating it with matching items found in obj, or creating one (while taking care of type)
- for remaining ones, I am deleting from source